### PR TITLE
feat: add BatchCrypto support in EDV

### PR DESCRIPTION
**Pull request by Baha-sk**
This change includes EDV batch Crypto computation to help support combining all MAC/Encryption/KW operations in one 1 call

Part of #3175

Signed-off-by: Baha Shaaban <baha.shaaban@securekey.com>

### CODE REVIEW

File: component/storage/edv/encryptedformatter.go [hunk 1]
```diff
@@ -126,18 +142,46 @@ func (e *EncryptedFormatter) format(keyAndTagPrefix, key string, value []byte, t
 
 	var formattedValue []byte
 
-	if documentID != "" {
```
Comment (DRK3) [inline component/storage/edv/encryptedformatter.go:129]: Looks like you removed the `if documentID != ""` check - was that intentional? I can't remember why I put that there in the first place, so if that was an intentional change we should discuss briefly just to make sure it doesn't break something

Comment (Baha-sk) [reply component/storage/edv/encryptedformatter.go:129]:
```
	if e.useDeterministicDocumentIDs {
		documentID, err = e.generateDeterministicDocumentID(keyAndTagPrefix, key)
		if err != nil {
			return "", nil, nil, fmt.Errorf("failed to format key into an encrypted document ID: %w", err)
		}
	} else {
		documentID, err = generateRandomDocumentID()
		if err != nil {
			return "", nil, nil, fmt.Errorf("failed to generate EDV-compatible ID: %w", err)
		}
	}
```

above will ensure `documentID` is not empty, unless one of those calls return an empty value, which is not the case.

Comment (DRK3) [reply component/storage/edv/encryptedformatter.go:129]: Makes sense!

**Review by DRK3 — APPROVED** [review]

File: component/storage/edv/encryptedformatter.go (hunk 1 above)
Comment (Baha-sk) [reply component/storage/edv/encryptedformatter.go]: reverted this change to minimize impact of this pr

**Comment by Baha-sk** [comment]
> I see [scripts/start_edv_test_docker_images.sh](https://github.com/hyperledger/aries-framework-go/pull/3189/files#diff-e924216c475870598d056b7998f70d374bce2323a280fe61813390a55f9b9f80) in the changelist but no actual line changes in the diff - was something actually changed?

actually, it's just security access to this file to be executable, i just made it accessible for local execution, current mode was read/write only, now it's also executable. Notice the `chomd` mode udpate: `100644 → 100755`

File: component/storage/edv/encryptedformatter.go (hunk 1 above)
Comment (DRK3) [reply component/storage/edv/encryptedformatter.go]: Actually on closer look, `generateDeterministicDocumentID` can return a blank ID, so yeah reverting is the right thing to do

**Review by fqutishat — APPROVED** [review]

**Pull request merged**